### DISCUSSION TITLE
issue: After updating react-hook-form input are not getting registerd

### DISCUSSION THREAD

**Discussion by vikas990** [react-hook-form/react-hook-form#7432] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/friendly-microservice-wq4w8?file=/components/Form.js:631-638

### Steps to reproduce

1. clone the repo 'https://github.com/vikas990/Mathrithms-test.git'
2. run "npm install"
3. run "npm run dev"
4. See error:-
<img width="1680" alt="Screenshot 2022-01-04 at 4 02 41 PM" src="https://user-images.githubusercontent.com/50900838/148046070-4752c35f-79ba-41bb-95d3-0e61af140d5e.png">

### Expected behaviour

the Expected behaviour is that it should submit the value.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
your csb doens't work

<img width="1085" alt="Screen Shot 2022-01-04 at 10 29 38 pm" src="https://user-images.githubusercontent.com/10513364/148052692-fad3ebd3-7568-449e-a967-c6b5f220e3d5.png">

from your screenshot it looks like you have a email input and the input is not valid.

**Reply by vikas990** [reply]
I don't know why CSD not working but the problem is when I try to submit data the values in the field did not register which leads to an error. As I have shared above.
**Form.js**
```
import { Card } from "./main.styles";
import { useForm, useWatch } from "react-hook-form";
import { Button, FormControl, InputField, MenuItem, Select } from "./utils";
const Controller = ({ control, register, name, rules, render }) => {
  const value = useWatch({
    control,
    name,
  });
  const props = register(name);
  return render({
    value,
    onChange: (e) =>
      props.onChange({
        target: {
          name,
          value: e.target.value,
        },
      }),

    onBlur: props.onBlur,
    name: props.name,
  });
};
export default function Form() {
  const {
    register,
    handleSubmit,
    reset,
    control,
    formState: { errors },
  } = useForm();
  console.log(errors);
  const submitHandler = (data) => {
    console.log(data);
  };
  return (
    <div>
      <Card url={v}>
        <h1>Form</h1>
        <form onSubmit={handleSubmit(submitHandler)}>
          <InputField
            fullWidth
            placeholder="name"
            name="name"
            register={register}
          />
          <p style={{ color: "red" }}>{errors?.name?.message}</p>
          <br />
          <br />
          <Controller
            {...{
              control,
              register,
              name: "dropdown",
              rules: {},
              render: (props) => (
                <FormControl fullWidth>
                  <Select {...props}>
                    <MenuItem value="one">one</MenuItem>
                    <MenuItem value="two">Two</MenuItem>
                    <MenuItem value="three">Three</MenuItem>
                  </Select>
                </FormControl>
              ),
            }}
          />

          <br />
          <br />
          <Button fullWidth type="submit">
            Submit
          </Button>
        </form>
      </Card>
    </div>
  );
}

```

Thank You

**Reply by vikas990** [reply]
i have updated the CSD please :- https://codesandbox.io/s/youthful-morning-t6n7e?file=/components/Form.js

**Comment by bluebill1049** [comment]
<img width="906" alt="Screen Shot 2022-01-05 at 9 10 20 am" src="https://user-images.githubusercontent.com/10513364/148130644-3edd71ff-4425-4658-aa26-1f612db614ee.png">

can you share steps to reproduce the issue?

**Reply by vikas990** [reply]
hi thanks for the reply. please check the git repo for that

clone the repo 'https://github.com/vikas990/Mathrithms-test.git'
run "npm install"
run "npm run dev"

thank you
